sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3846,6 +3846,24 @@ def teardown_no_lock(self,
 
             # To avoid undefined local variables error.
             stdout = stderr = ''
+        elif terminate and isinstance(cloud, clouds.Kubernetes):
+            # Here we handle termination of Kubernetes by ourselves instead of
+            # Ray autoscaler.
```

Comment:
The logic is similar to [remaining_nodes()](https://github.com/ray-project/ray/blob/ray-2.4.0/python/ray/autoscaler/_private/commands.py#L438) method.